libp2p/muxers/mplex/mplex.nim

```diff
@@ -163,24 +184,36 @@ method newStream*(m: Mplex,
   result = newConnection(channel)
   result.peerInfo = m.connection.peerInfo
 
-method close*(m: Mplex) {.async, gcsafe.} =
-  trace "closing mplex muxer"
-
-  if not m.connection.closed():
-    await m.connection.close()
+  asyncCheck m.cleanupChann(channel)
```

Comment:
Because `newStream` needs to return, so if I `await` here, I'd never return and if you look at `cleanupChan` it actually waits for the close event, so it is a sort of finalizer and doesn't execute until the event triggers.